test/test_prune_restarts.py

```diff
@@ -222,3 +222,38 @@ def test_prune_restarts(restarts,
     ]
 
     assert restarts_to_prune_indices == expected_restart_indices
+
+
+def test_prune_restarts_ignores_empty_restart_dirs():
+    # Test that in date-based restart pruning ignores any empty restart dirs
+
+    write_test_config(restart_freq='1YS')
+
+    # Create restart files
+    restart_datetimes = [(4, "1903-01-01 00:00:00"),
+                         (5, "1903-06-01 00:00:00"),
+                         (6, "1905-01-01 00:00:00")]
+    for index, datetime in restart_datetimes:
+        make_ocean_restart_dir(start_dt="1900-01-01 00:00:00",
+                               run_dt=datetime,
+                               calendar=4,
+                               restart_index=index,
+                               additional_path='ocean')
+
+    # Create empty restart dirs
+    make_expt_archive_dir(type='restart', index=0)
+    make_expt_archive_dir(type='restart', index=2)
+
+    with cd(ctrldir):
+        lab = payu.laboratory.Laboratory(lab_path=str(labdir))
+        expt = payu.experiment.Experiment(lab, reproduce=False)
+
+        # Function to test
+        restarts_to_prune = expt.get_restarts_to_prune()
```

Comment:
We don't use warnings in `payu` (and probably should) then it would be simple to check that this emitted the warning you added.

https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns

I'm not suggesting you add this, just sayin'

Reply:
Feel free to resolve this convo once you've had a chance to read it.

Reply:
Thanks, that is good to know about warnings. I'll look into adding it with future work